fmriprep/interfaces/bids.py

```diff
@@ -201,9 +203,10 @@ def __init__(self, out_path_base=None, **inputs):
     def _run_interface(self, runtime):
         src_fname, _ = _splitext(self.inputs.source_file)
         _, ext = _splitext(self.inputs.in_file[0])
-        compress = ext == '.nii'
-        if compress:
+        if self.inputs.compress is True:
             ext = '.nii.gz'
+        elif self.inputs.compress is False:
+            ext = '.nii'
```

Comment:
just a note: gifti files can be compressed, but the compression is inside the file in the data blob, rather than external. (i do think this sort of logic should make it's way to nibabel, with perhaps a `compress=???` option in to_filename).